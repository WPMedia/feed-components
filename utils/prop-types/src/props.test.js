import { generatePropsForFeed } from './'
import PropTypes from 'fusion:prop-types'

it('generates proptypes for sitemap', () => {
  const shape = generatePropsForFeed('sitemap', PropTypes)
  expect(JSON.parse(PropTypes.stringify(shape))).toMatchSnapshot()
})

it('omits specified properties from sitemap propypes', () => {
  const shape = generatePropsForFeed('sitemap', PropTypes, ['includePromo'])
  expect(JSON.parse(PropTypes.stringify(shape))).toMatchSnapshot()
})
