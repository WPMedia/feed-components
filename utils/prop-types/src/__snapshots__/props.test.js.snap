// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates proptypes for rss 1`] = `
Object {
  "channelCategory": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "Category that describes this RSS feed, if blank it will be excluded",
      "group": "Channel",
      "label": "Category",
    },
    "type": "string",
  },
  "channelCopyright": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "RSS Copyright value otherwise it will be excluded",
      "group": "Channel",
      "label": "Copyright",
    },
    "type": "string",
  },
  "channelDescription": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "RSS Channel Description, defaults to website name + \\" News Feed\\"",
      "group": "Channel",
      "label": "RSS Description",
    },
    "type": "string",
  },
  "channelLogo": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "URL to the logo for this RSS feed",
      "group": "Channel",
      "label": "Logo URL",
    },
    "type": "string",
  },
  "channelTTL": Object {
    "tags": Object {
      "defaultValue": "1",
      "description": "Number of minutes to wait to check for new content, defaults to 1",
      "group": "Channel",
      "label": "Time To Live",
    },
    "type": "string",
  },
  "channelTitle": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "RSS Channel Title, defaults to website name",
      "group": "Channel",
      "label": "RSS Title",
    },
    "type": "string",
  },
  "channelUpdateFrequency": Object {
    "tags": Object {
      "defaultValue": "1",
      "description": "Number of Update Periods to wait to check for new content, defaults to 1",
      "group": "Channel",
      "label": "Update Frequency",
    },
    "type": "string",
  },
  "channelUpdatePeriod": Object {
    "args": Array [
      "hourly",
      "daily",
      "weekly",
      "monthly",
      "yearly",
    ],
    "tags": Object {
      "defaultValue": "hourly",
      "description": "Which period of time should be used, defaults to hourly",
      "group": "Channel",
      "label": "Update Period",
    },
    "type": "oneOf",
  },
  "imageCaption": Object {
    "tags": Object {
      "defaultValue": "caption",
      "description": "ANS value for associated story image used for the <media:caption> sitemap tag, defaults to caption",
      "group": "Featured Image",
      "label": "ANS image caption key",
    },
    "type": "string",
  },
  "imageCredits": Object {
    "tags": Object {
      "defaultValue": "credits.by[].name",
      "description": "ANS value for associated story image credits for the <media:credits> sitemap tag, defaults to credits.by[].name",
      "group": "Featured Image",
      "label": "ANS image credits key",
    },
    "type": "string",
  },
  "imageTitle": Object {
    "tags": Object {
      "defaultValue": "title",
      "description": "ANS value for associated story used for the <media:title> sitemap tag, defaults to title",
      "group": "Featured Image",
      "label": "ANS image title key",
    },
    "type": "string",
  },
  "includeContent": Object {
    "args": Array [
      "0",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
      "8",
      "9",
      "10",
      "all",
    ],
    "tags": Object {
      "defaultValue": "all",
      "description": "Number of paragraphs to include, defaults to all",
      "group": "Item",
      "label": "Number of paragraphs to include",
    },
    "type": "oneOf",
  },
  "includePromo": Object {
    "tags": Object {
      "defaultValue": true,
      "description": "Include the featured image in <media:content>",
      "group": "Featured Image",
      "label": "Include promo images",
    },
    "type": "bool",
  },
  "itemCategory": Object {
    "tags": Object {
      "defaultValue": "",
      "description": "ANS field to use for article category, if blank will be excluded",
      "group": "Item",
      "label": "Category",
    },
    "type": "string",
  },
  "itemCredits": Object {
    "tags": Object {
      "defaultValue": "credits.by[].name",
      "description": "ANS value for credits in the <dc:creator> tag, will join multiple names with a comma. defaults to credits.by[].name",
      "group": "Item",
      "label": "ANS credits key",
    },
    "type": "string",
  },
  "itemDescription": Object {
    "tags": Object {
      "defaultValue": "description.basic",
      "description": "ANS fields to use for article description, defaults to description.basic",
      "group": "Item",
      "label": "Description",
    },
    "type": "string",
  },
  "itemTitle": Object {
    "tags": Object {
      "defaultValue": "headlines.basic",
      "description": "ANS fields to use for article title, defaults to headlines.basic",
      "group": "Item",
      "label": "Title",
    },
    "type": "string",
  },
  "pubDate": Object {
    "args": Array [
      "created_date",
      "display_date",
      "first_publish_date",
      "last_updated_date",
      "publish_date",
    ],
    "tags": Object {
      "defaultValue": "display_date",
      "description": "Which date field should be used, defaults to display_date",
      "group": "Item",
      "label": "Publication Date",
    },
    "type": "oneOf",
  },
  "resizerKVP": Object {
    "tags": Object {
      "defaultValue": Object {
        "height": 0,
        "width": 0,
      },
      "description": "Optional height and or width to resize all images to",
      "group": "Featured Image",
      "label": "Image height and or width",
    },
    "type": "kvp",
  },
}
`;

exports[`generates proptypes for sitemap 1`] = `
Object {
  "changeFreq": Object {
    "args": Array [
      "always",
      "hourly",
      "daily",
      "weekly",
      "monthly",
      "yearly",
      "never",
      "Exclude from sitemap",
    ],
    "tags": Object {
      "defaultValue": "always",
      "description": "What is the Change frequency of the sitemap",
      "group": "Format",
      "label": "change frequency",
    },
    "type": "oneOf",
  },
  "includePromo": Object {
    "tags": Object {
      "defaultValue": true,
      "description": "Include an image in the sitemap",
      "group": "Format",
      "label": "Include promo images?",
    },
    "type": "bool",
  },
  "lastMod": Object {
    "args": Array [
      "created_date",
      "display_date",
      "first_publish_date",
      "last_updated_date",
      "publish_date",
    ],
    "tags": Object {
      "defaultValue": "last_updated_date",
      "description": "Which date field should be used in the sitemap",
      "group": "Format",
      "label": "Last Modified Date",
    },
    "type": "oneOf",
  },
  "priority": Object {
    "args": Array [
      "0.0",
      "0.1",
      "0.2",
      "0.3",
      "0.4",
      "0.5",
      "0.6",
      "0.7",
      "0.8",
      "0.9",
      "1.0",
      "Exclude from sitemap",
    ],
    "tags": Object {
      "defaultValue": "0.5",
      "description": "What is the priority of the sitemap",
      "group": "Format",
      "label": "priority",
    },
    "type": "oneOf",
  },
  "resizerKVP": Object {
    "tags": Object {
      "defaultValue": Object {
        "height": 0,
        "width": 0,
      },
      "description": "Optional height and or width to resize all images to",
      "group": "Format",
      "label": "Image height and or width",
    },
    "type": "kvp",
  },
}
`;

exports[`omits specified properties from sitemap propypes 1`] = `
Object {
  "changeFreq": Object {
    "args": Array [
      "always",
      "hourly",
      "daily",
      "weekly",
      "monthly",
      "yearly",
      "never",
      "Exclude from sitemap",
    ],
    "tags": Object {
      "defaultValue": "always",
      "description": "What is the Change frequency of the sitemap",
      "group": "Format",
      "label": "change frequency",
    },
    "type": "oneOf",
  },
  "lastMod": Object {
    "args": Array [
      "created_date",
      "display_date",
      "first_publish_date",
      "last_updated_date",
      "publish_date",
    ],
    "tags": Object {
      "defaultValue": "last_updated_date",
      "description": "Which date field should be used in the sitemap",
      "group": "Format",
      "label": "Last Modified Date",
    },
    "type": "oneOf",
  },
  "priority": Object {
    "args": Array [
      "0.0",
      "0.1",
      "0.2",
      "0.3",
      "0.4",
      "0.5",
      "0.6",
      "0.7",
      "0.8",
      "0.9",
      "1.0",
      "Exclude from sitemap",
    ],
    "tags": Object {
      "defaultValue": "0.5",
      "description": "What is the priority of the sitemap",
      "group": "Format",
      "label": "priority",
    },
    "type": "oneOf",
  },
  "resizerKVP": Object {
    "tags": Object {
      "defaultValue": Object {
        "height": 0,
        "width": 0,
      },
      "description": "Optional height and or width to resize all images to",
      "group": "Format",
      "label": "Image height and or width",
    },
    "type": "kvp",
  },
}
`;
